trollflow2/plugins/__init__.py

```diff
@@ -133,23 +130,36 @@ def save_datasets(job):
     objs = []
     base_config = job['input_mda'].copy()
     base_config.pop('dataset', None)
+    renames = {}
     for fmat, fmat_config in plist_iter(job['product_list']['product_list'], base_config):
         fname_pattern = fmat['fname_pattern']
         filename = compose(os.path.join(fmat['output_dir'], fname_pattern), fmat)
+        directory = fmat['output_dir']
+        if not os.path.exists(directory):
+            os.makedirs(directory)
+        if fmat.get('use_tmp_file', False):
```

Comment:
Short explanation here would be nice: what's the overall purpose of the temporary file. Something like "Save file to a temporary file and rename after writing so that partly written files will not be accessed"

Reply:
thanks, I'll fix that